control-plane/controllers/resources/gateway_controller_setup.go

```diff
@@ -0,0 +1,137 @@
+		Owns(&corev1.ServiceAccount{}).
+		Watches(
+			source.NewKindWithCache(&meshv2beta1.GatewayClass{}, mgr.GetCache()),
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				gc := o.(*meshv2beta1.GatewayClass)
+				if gc == nil {
+					return nil
+				}
+
+				gateways, err := getGatewaysReferencingGatewayClass[L](context.Background(), k8sClient, gc.Name, index)
+				if err != nil {
+					return nil
+				}
+
+				return gateways.ReconcileRequests()
+			})).
+		Watches(
+			source.NewKindWithCache(&meshv2beta1.GatewayClassConfig{}, mgr.GetCache()),
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				gcc := o.(*meshv2beta1.GatewayClassConfig)
+				if gcc == nil {
+					return nil
+				}
+
+				classes, err := getGatewayClassesByGatewayClassConfigName(context.Background(), k8sClient, gcc.Name)
+				if err != nil {
+					return nil
+				}
+
+				var requests []reconcile.Request
+				for _, class := range classes.Items {
+					if class == nil {
+						continue
+					}
+
+					gateways, err := getGatewaysReferencingGatewayClass[L](context.Background(), k8sClient, class.Name, index)
+					if err != nil {
+						continue
+					}
+
+					requests = append(requests, gateways.ReconcileRequests()...)
+				}
+
+				return requests
+			})).
+		Complete(gwc)
+}
+
+// TODO: uncomment when moving the CRUD hooks from mesh gateway controller
+//func getGatewayClassConfigByGatewayClassName(ctx context.Context, k8sClient client.Client, className string) (*meshv2beta1.GatewayClassConfig, error) {
+//	gatewayClass, err := getGatewayClassByName(ctx, k8sClient, className)
+//	if err != nil {
+//		return nil, err
+//	}
+//
+//	if gatewayClass == nil {
+//		return nil, nil
+//	}
+//
+//	gatewayClassConfig := &meshv2beta1.GatewayClassConfig{}
+//	if ref := gatewayClass.Spec.ParametersRef; ref != nil {
+//		if ref.Group != meshv2beta1.MeshGroup || ref.Kind != v2beta1.KindGatewayClassConfig {
+//			// TODO @Gateway-Management additionally check for controller name when available
+//			return nil, nil
+//		}
+//
+//		if err := k8sClient.Get(ctx, types.NamespacedName{Name: ref.Name}, gatewayClassConfig); err != nil {
+//			return nil, client.IgnoreNotFound(err)
+//		}
+//	}
+//	return gatewayClassConfig, nil
+//}
+//
+//func getGatewayClassByName(ctx context.Context, k8sClient client.Client, className string) (*meshv2beta1.GatewayClass, error) {
+//var gatewayClass meshv2beta1.GatewayClass
+//
+//	if err := k8sClient.Get(ctx, types.NamespacedName{Name: className}, &gatewayClass); err != nil {
+//		return nil, client.IgnoreNotFound(err)
+//	}
+//	return &gatewayClass, nil
+//}
+
+// getGatewayClassesByGatewayClassConfigName queries all GatewayClass resources in the
+// cluster and returns any that reference the given GatewayClassConfig.
+func getGatewayClassesByGatewayClassConfigName(ctx context.Context, k8sClient client.Client, configName string) (*meshv2beta1.GatewayClassList, error) {
```

Comment:
For consistency with the below

```suggestion
// getGatewayClassesReferencingGatewayClassConfig queries all GatewayClass resources in the
// cluster and returns any that reference the given GatewayClassConfig by name.
func getGatewayClassesReferencingGatewayClassConfig(ctx context.Context, k8sClient client.Client, configName string) (*meshv2beta1.GatewayClassList, error) {
```